rust/rope/src/spans.rs

```diff
@@ -301,21 +301,33 @@ impl<T: Clone> Spans<T> {
         *self = b.build();
     }
 
-    // FIXME: Instead of iterating through all spans every time, another option would be to go
-    // leaf-by-leaf, and check each leaf for whether or not it has any items in the interval;
-    // if they don't we keep them unchanged, otherwise we do this operation, but only within the leaf.
-    //
     /// Deletes all spans that intersect with `interval`.
     pub fn delete_intersecting(&mut self, interval: Interval) {
-        let mut builder = SpansBuilder::new(self.len());
-        for (iv, data) in self.iter() {
-            // check if spans overlaps with interval
-            if iv.intersect(interval).is_empty() {
-                // keep the ones that are not overlapping
-                builder.add_span(iv, data.clone());
+        let mut b = TreeBuilder::new();
+        // the end of the last leaf seen
+        let mut end_offset = 0;
+        let mut leaf_iterator = self.iter_leaf_nodes();
+        while let Some((node, leaf)) = leaf_iterator.next() {
+            let SpansLeaf { len, spans } = leaf;
```

Comment:
CI chokes on this

Reply:
Thanks! However this PR will not pass CI for now, there is a logical bug somewhere preventing this from working on extremely large sizes. IDK if it is a problem with spans or this particular feature.